File: Project/backend/app/models/agenda_utente.py
from datetime import datetime
from sqlalchemy import Column, DateTime
from sqlmodel import Field, SQLModel

class AgendaUtenteBase(SQLModel):
    id_utente: int = Field(foreign_key="utente.id")
    id_poi: int = Field(foreign_key="poi.id")
    titolo: str = Field(max_length=100)
    orario_inizio: datetime = Field(sa_column=Column(DateTime(timezone=True)))
    orario_fine: datetime = Field(sa_column=Column(DateTime(timezone=True)))

class AgendaUtente(AgendaUtenteBase, table=True):
    __tablename__ = "agenda_utente"
    id: int | None = Field(default=None, primary_key=True, index=True)

class AgendaUtenteCreate(AgendaUtenteBase):
    pass

class AgendaUtentePublic(AgendaUtenteBase):
    id: int

class AgendaUtenteUpdate(SQLModel):
    #id_utente: int | None = Field(default=None)
    id_poi: int | None = Field(default=None)
    titolo: str | None = Field(default=None, max_length=100)
    orario_inizio: datetime | None = Field(default=None)
    orario_fine: datetime | None = Field(default=None)

class AgendaUtenteContext(AgendaUtentePublic):
    distanza_metri: float
    avviso: str

File: Project/ContextAware_app/app/src/main/java/com/example/myapplication/data/model/agenda_utente.kt
package com.example.myapplication.data.model

import kotlinx.serialization.Serializable


@Serializable
data class AgendaUtenteCreate(
    val id_utente: Int,
    val id_poi: Int,
    val titolo: String,
    val orario_inizio: String,
    val orario_fine: String
)

@Serializable
data class AgendaUtentePublic(
    val id: Int,
    val id_utente: Int,
    val id_poi: Int,
    val titolo: String,
    val orario_inizio: String,
    val orario_fine: String
)


@Serializable
data class AgendaUtenteUpdate(
    val id_poi: Int? = null,
    val titolo: String? = null,
    val orario_inizio: String? = null,
    val orario_fine: String? = null
)


@Serializable
data class AgendaUtenteContext(
    val id: Int,
    val id_utente: Int,
    val id_poi: Int,
    val titolo: String,
    val orario_inizio: String,
    val orario_fine: String,
    val distanza_metri: Double,
    val avviso: String
)